src/js/utils/validator.ts

```diff
@@ -75,7 +75,7 @@ export function typeOf(value) {
  * @param {function} predicate - The function evaluating whether the property has a valid value and can be considered compliant. Inputs are the object and its key.
  * @returns {boolean} Does the customObj have at least the same keys as defaultObj, and do their properties also share the same keys ?
  */
-export function isDeepKeyCompliant(defaultObj, customObj, predicate) {
+export function isDeepKeyCompliant(defaultObj: {[key: string]: any}, customObj: {[key: string]: any}, predicate: (x: string, y: object) => boolean): boolean {
```

Comment:
What's the issue with using standard typing here? ie: 

`export function isDeepKeyCompliant(defaultObj: object, customObj: object, predicate: (x: string, y: object) => boolean): boolean`

Reply:
Currently, the `noImplicitAny` flag is enabled, so you need to explicitly specify what the index/key type for an object is.
I kept all the strict settings enabled at least for now (They are also enabled by default). Can definitely discuss to determine what we want to keep.

Reply:
To note, there is the '[suppressImplicitAnyIndexErrors'](https://www.typescriptlang.org/docs/handbook/compiler-options.html) tsconfig compiler option

Reply:
we should create an interface or type for the generic object `{[key: string]: any},` as i can see it being used quite a lot

Reply:
bump this comment